### PR TITLE
[Delivers #92792448] list all proposals that the user is involved in

### PR DESCRIPTION
**Pull request by afeld**
...on `/carts`. Includes Proposals for which the users is a:

* Requester
* Approver
* Delegate
* Observer

### CODE REVIEW

File: app/policies/proposal_policy.rb [hunk 1]
```diff
@@ -79,9 +79,19 @@ def resolve
...
+        -- approver
         OR EXISTS (SELECT id FROM approvals
                    WHERE proposal_id = proposals.id AND user_id = :user_id)
+        -- delegate
+        OR EXISTS (
+          SELECT approvals.id FROM approvals
+          LEFT OUTER JOIN approval_delegates
+          ON approval_delegates.assigner_id = approvals.user_id
+          WHERE approval_delegates.assignee_id = :user_id
+        )
```
Comment (afeld) [inline app/policies/proposal_policy.rb]: _Somehow_ got this right on the first try. `</humblebrag>`

skeletons dance

Comment (cmc333333) [inline app/policies/proposal_policy.rb]: It looks like there's not restriction by proposal? I think all that needs to change is to add:
```sql
AND approvals.proposal_id = proposals.id
```

Comment (afeld) [inline app/policies/proposal_policy.rb]: Way to burst my bubble, @cmc333333.

**Comment by afeld** [comment]
Updated! Fixed the SQL condition issue...not sure how/if to write a test for it though.

**Pull request merged**